### ISSUE TITLE
If the system has the wrong time, things will start to fail

### ISSUE THREAD

**Issue by Itxaka** [rancher-sandbox/cOS-toolkit#1359]
If your system has the wrong time, it will start failing in certificates, and other stuff. We currently have nothing in place to sync time :o

Create a chronyd package? or just make it part of the distro/base?

**Comment by mudler** [comment]
I'm not sure if we should do anything in there as it is more or less depending on what you put on a derivative. c3os for example uses systemd-timesyncd which is there out of the box

**Comment by mudler** [comment]
Yes it uses zipper indeed, in teal I think that or we either document a way to get few packages in, or we do offer some default. I say this because it won't be just NTPD here, but same reasoning could happen for other services ( let's say, for example iscsid?)

So in that case we should ask to add timesyncd to the base image IMHO instead. In that way the image comes with already a stable default ( at the end of the day teal is an opinionated image from sle micro that is built just for us ) although I see your point of offering an alternative there, I'm just not sure that it is needed for teal

**Comment by Itxaka** [comment]
Oh hell yeah, if we can get the base image to come with it, I'm all for it. It just seems to me that is such a basic service, especially with security in mind, certificates and so on, that it just needs to be there. I.e. docker images may fail to download, nodes fail to register, etc.. if time is drifted, so for me it seems a must have.

Having as a package also allows for more flexibility, i.e. especially if optional. But I really think is something that has to be fixed or that we need to facilitate to the end user/derivatives

**Comment by mudler** [comment]
Just checked, and systemd-timesyncd is indeed part of the image, we just need to enable it on boot:

```
docker run --rm -ti registry.suse.com/suse/sle-micro-rancher/5.2 /bin/bash -c "cat /usr/lib/systemd/system/systemd-timesyncd.service"
```

**Comment by kkaempf** [comment]
Keep in mind that if anything is missing from Teal, we can add it :wink:
